app/src/main/java/com/github/se/travelpouch/helper/FileDownloader.kt

```diff
@@ -0,0 +1,55 @@
+package com.github.se.travelpouch.helper
+
+import android.content.ContentResolver
+import android.util.Log
+import androidx.documentfile.provider.DocumentFile
+import com.google.firebase.storage.FirebaseStorage
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+/** Helper class to download files from Firbase storage to local files */
+open class FileDownloader(private val contentResolver: ContentResolver) {
+  /**
+   * Download a file described by source and store it in the folder pointed by destinationFolder.
+   *
+   * @param sourceMimeType The mimeType of the source file
+   * @param sourceTitle The title of the source file
+   * @param sourceRef The reference of the source file
+   * @param destinationFolder A pointer to the folder in which to store the file
+   */
+  fun downloadFile(
+      sourceMimeType: String,
+      sourceTitle: String,
+      sourceRef: String,
+      destinationFolder: DocumentFile
+  ) {
+    val file = destinationFolder.createFile(sourceMimeType, sourceTitle)
+
+    if (file == null) {
+      Log.e("DocumentViewModel", "Failed to create document file in specified directory")
+      return
+    }
+
+    val storageRef = FirebaseStorage.getInstance("gs://travelpouch-7d692.appspot.com").reference
```

Comment:
Just a note for later: once our general use of constants is settled, it would be useful to group them as properties

Reply:
Yes good idea ! As we will change some things with the download